core/src/main/java/org/infinispan/statetransfer/StateTransferLock.java

```diff
@@ -39,15 +43,35 @@
    // transaction data latch
    void notifyTransactionDataReceived(int topologyId);
 
-   void waitForTransactionData(int expectedTopologyId, long timeout, TimeUnit unit) throws InterruptedException;
+   CompletableFuture<Void> transactionDataFuture(int expectedTopologyId);
+
+   @Deprecated
+   default void waitForTransactionData(int expectedTopologyId, long timeout, TimeUnit unit) throws InterruptedException, TimeoutException {
```

Comment:
This isn't an SPI, so I'm not sure it's worth putting the implementation in the interface.